### PR TITLE
Improve normalization close error message

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Improve the error message when the normalization process fails

## How
* Separate the two cases that cause an error when attempting to close the normalization process
* Include more details in the error message

## Recommended reading order
1. `DefaultNormalizationRunner.java`

## Tests

* All existing tests pass
* Project builds successfully locally

### CODE REVIEW

**Review by gosusnp — APPROVED** [review]

**Review by alovew — APPROVED** [review]

**Review by colesnodgrass — APPROVED** [review]

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java [hunk 1]
```diff
@@ -194,10 +194,20 @@ public void close() throws Exception {
...
-    if (process.isAlive() || process.exitValue() != 0) {
-      throw new WorkerException("Normalization process wasn't successful");
+
+    /*
+     * After attempting to close the process check the following:
+     *
+     * Did the process actually terminate? If "yes", did it do so nominally?
+     */
+    if (process.isAlive()) {
+      throw new WorkerException("Normalization process " + process.pid() + " wasn't successful");
```
Comment (edgao) [inline airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java:206]: is it more accurate to say that we didn't successfully  terminate it?

**Review by edgao — APPROVED** [review]
one question about log message wording, otherwise lgtm

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java (hunk 1 above)
Comment (jdpgrailsdev) [reply airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java:206]: @edgao Definitely.  Will fix.

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java [hunk 2]
```diff
@@ -194,10 +194,20 @@ public void close() throws Exception {
       return;
     }
 
-    LOGGER.debug("Closing normalization process");
+    LOGGER.debug("Terminating normalization process {}...", process.pid());
```
Comment (benmoriceau) [inline airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java:197]: Should we move that to info?

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java [hunk 3]
```diff
@@ -194,10 +194,20 @@ public void close() throws Exception {
...
+     * After attempting to close the process check the following:
+     *
+     * Did the process actually terminate? If "yes", did it do so nominally?
+     */
+    if (process.isAlive()) {
+      throw new WorkerException("Normalization process " + process.pid() + " did not terminate after 1 minute.");
+    } else if (process.exitValue() != 0) {
+      throw new WorkerException("Normalization process " + process.pid() + " did not terminate normally (exit code: " + process.exitValue() + ")");
+    } else {
+      LOGGER.debug("Normalization process {} successfully terminated.", process.pid());
```
Comment (benmoriceau) [inline airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java:210]: Same comment about the log level

**Review by benmoriceau — APPROVED** [review]

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java (hunk 2 above)
Comment (jdpgrailsdev) [reply airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java:197]: @benmoriceau We could.  I'm not sure why it was debug to begin with.

Comment (jdpgrailsdev) [reply airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java]: Done

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java (hunk 1 above)
Comment (jdpgrailsdev) [reply airbyte-commons-worker/src/main/java/io/airbyte/workers/normalization/DefaultNormalizationRunner.java]: Done

**Comment by Aggiekev** [comment]
I believe this merge is causing my Connections to fail during normalization. The error message isn't descriptive in the logs but this is the only merge that dealt with normalization in the last release which is where my Connections are failing. They fail at different points in the transformation for every single connection. My guess is that it is related to the waiting for one minute to throw the error?

**Comment by edgao** [comment]
@jdpgrailsdev do you recognize what's happening in that sync? It looks like normalization terminates cleanly (`2022-12-06 05:50:16 normalization > Completed successfully`) but DefaultNormalizationWorker thinks it failed:
```
2022-12-06 05:50:17 INFO i.a.w.t.TemporalAttemptExecution(lambda$getWorkerThread$5):198 - Completing future exceptionally...
io.airbyte.workers.exception.WorkerException: Normalization Failed.
        at io.airbyte.workers.general.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:103) ~[io.airbyte-airbyte-commons-worker-0.40.23.jar:?]
        at io.airbyte.workers.general.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:34) ~[io.airbyte-airbyte-commons-worker-0.40.23.jar:?]
        at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$5(TemporalAttemptExecution.java:195) ~[io.airbyte-airbyte-workers-0.40.23.jar:?]
        at java.lang.Thread.run(Thread.java:1589) ~[?:?]
```

**Comment by jdpgrailsdev** [comment]
> @jdpgrailsdev do you recognize what's happening in that sync? It looks like normalization terminates cleanly (`2022-12-06 05:50:16 normalization > Completed successfully`) but DefaultNormalizationWorker thinks it failed:
> 
> ```
> 2022-12-06 05:50:17 INFO i.a.w.t.TemporalAttemptExecution(lambda$getWorkerThread$5):198 - Completing future exceptionally...
> io.airbyte.workers.exception.WorkerException: Normalization Failed.
>         at io.airbyte.workers.general.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:103) ~[io.airbyte-airbyte-commons-worker-0.40.23.jar:?]
>         at io.airbyte.workers.general.DefaultNormalizationWorker.run(DefaultNormalizationWorker.java:34) ~[io.airbyte-airbyte-commons-worker-0.40.23.jar:?]
>         at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$5(TemporalAttemptExecution.java:195) ~[io.airbyte-airbyte-workers-0.40.23.jar:?]
>         at java.lang.Thread.run(Thread.java:1589) ~[?:?]
> ```

@edgao There was an issue introduced before this PR that caused a failure when attempting to stop/close the process.  This was due to calling `.pid()` on the `KubePodProcess`, which had not implemented the method.  This PR added logging of errors on line 83, which may give us more information about what exactly failed.

**Pull request merged**